src/ProducerConfig.cc

```diff
@@ -224,6 +250,18 @@ ProducerConfig::ProducerConfig(const Napi::Object& producerConfig) : topic("") {
   if (PRODUCER_BATCHING_TYPE.count(batchingType)) {
     this->cProducerConfig.get()->conf.setBatchingType(PRODUCER_BATCHING_TYPE.at(batchingType));
   }
+
+  if (producerConfig.Has(CFG_MESSAGE_ROUTER)) {
```

Comment:
We can move this logic to 

```
    if (MESSAGE_ROUTING_MODE.count(messageRoutingMode)) {
      pulsar_producer_configuration_set_partitions_routing_mode(this->cProducerConfig.get(),
                                                                MESSAGE_ROUTING_MODE.at(messageRoutingMode));
    }
    if (messageRoutingMode == "CustomPartition") {
     // here
    }
```

only set it when `messageRoutingMode == CustomPartition`

And if (`mesageRoutingMode == CustomPartition` && `messageRouter == null`) need throw a error.